frontend/packages/console-shared/src/components/catalog/details/CatalogDetailsModal.tsx

```diff
@@ -33,43 +40,111 @@ const CatalogDetailsModal: React.FC<CatalogDetailsModalProps> = ({ item, onClose
+          onClick={handleClick}
+          data-test={`${name}-cta-${index}`}
+        >
+          {label}
+        </Button>
+      );
+    }
+
+    // If there's an href, render as a link
+    if (to) {
+      return (
+        <Link
+          data-test={`${name}-cta-${index}`}
+          key={index}
+          // TODO: remove className once no longer needed by existing tests
+          className={`pf-v6-c-button pf-m-${variant || 'primary'} co-catalog-page__overlay-action`}
+          to={to}
+          role="button"
+          onClick={onClose}
+        >
+          {label}
+        </Link>
+      );
+    }
+
+    return null;
+  };
+
   const modalHeader = (
-    <>
-      <CatalogItemHeader
-        className="co-catalog-page__overlay-header"
-        title={title || name}
-        vendor={vendor}
-        {...getIconProps(item)}
-      />
-      <Split className="odc-catalog-details-modal__header">
-        <SplitItem>
-          {to && (
-            <div className="co-catalog-page__overlay-actions">
-              <Link
-                className="pf-v6-c-button pf-m-primary co-catalog-page__overlay-action"
-                to={to}
-                role="button"
-                onClick={onClose}
-              >
-                {label}
-              </Link>
-            </div>
-          )}
-        </SplitItem>
-        <SplitItem>{badges?.length > 0 ? <CatalogBadges badges={badges} /> : undefined}</SplitItem>
-      </Split>
-    </>
+    <CatalogItemHeader
+      // TODO: remove className once no longer needed by existing tests
+      className="co-catalog-page__overlay-header"
+      title={title || name}
+      vendor={vendor}
+      {...getIconProps(item)}
+    />
   );
 
   return (
     <Modal
+      // TODO: remove className once no longer needed by existing tests
       className="co-catalog-page__overlay co-catalog-page__overlay--right"
       isOpen={!!item}
       onClose={onClose}
       aria-label={item.name}
       header={modalHeader}
     >
-      <CatalogDetailsPanel item={item} />
+      <Stack hasGutter>
+        <StackItem>
+          {/* TODO: remove className once no longer needed by existing tests */}
+          <Split className="odc-catalog-details-modal__header">
+            <SplitItem>
+              {ctaLinks.length > 0 && (
+                // TODO: remove className once no longer needed by existing tests
+                <div className="co-catalog-page__overlay-actions">
+                  {ctaLinks.map((ctaLink, index) => renderAction(ctaLink, index))}
+                </div>
+              )}
+            </SplitItem>
+            <SplitItem isFilled />
+            <SplitItem>
+              {badges?.length > 0 ? <CatalogBadges badges={badges} /> : undefined}
+            </SplitItem>
+          </Split>
+        </StackItem>
+        <StackItem>
+          <Divider />
+        </StackItem>
+        <StackItem>
+          <CatalogDetailsPanel item={item} />
+        </StackItem>
+      </Stack>
```

Comment:
can we add `data-test` attributes as well here so it's easier to get off of the `className`s?